bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/internal/SseItemStatesEventBuilder.java

```diff
@@ -110,71 +108,75 @@ public SseItemStatesEventBuilder(final BundleContext bundleContext, final @Refer
+                } else if (pattern != null) {
+                    // if it's not a transformation pattern, then it must be a format string
+
+                    if (state instanceof QuantityType quantityState) {
+                        // sanity convert current state to the item state description unit in case it was
+                        // updated in the meantime. The item state is still in the "original" unit while the
+                        // state description will display the new unit:
+                        Unit<?> patternUnit = UnitUtils.parseUnit(pattern);
+                        if (patternUnit != null && !quantityState.getUnit().equals(patternUnit)) {
+                            quantityState = quantityState.toInvertibleUnit(patternUnit);
+                        }
 
-                            // The following exception handling has been added to work around a Java bug with formatting
-                            // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
-                            // This also handles IllegalFormatConversionException, which is a subclass of
-                            // IllegalArgument.
-                            try {
-                                displayState = state.format(pattern);
-                            } catch (IllegalArgumentException e) {
-                                logger.debug(
-                                        "Unable to format value '{}' of item {} with format '{}': {}, displaying raw state",
-                                        state, item.getName(), pattern, e.getMessage());
-                                displayState = state.toString();
-                            }
+                        if (quantityState != null) {
+                            state = quantityState;
                         }
+                    } else if (state instanceof DateTimeType type) {
+                        // Translate a DateTimeType state to the local time zone
+                        try {
+                            state = type.toLocaleZone();
+                        } catch (DateTimeException e) {
+                        }
+                    }
+
+                    // The following exception handling has been added to work around a Java bug with formatting
+                    // numbers. See http://bugs.sun.com/view_bug.do?bug_id=6476425
+                    // This also handles IllegalFormatConversionException, which is a subclass of
+                    // IllegalArgument.
+                    try {
+                        displayState = state.format(pattern);
+                    } catch (IllegalArgumentException e) {
+                        logger.debug(
+                                "Unable to format value '{}' of item {} with format '{}': {}, displaying raw state",
+                                state, item.getName(), pattern, e.getMessage());
+                        displayState = state.toString();
                     }
```

Comment:
Is this still needed? According to https://bugs.openjdk.org/browse/JDK-6476425 this issue has been fixed in Java 7!

Reply:
We probably have at least to catch `IllegalFormatConversionException`. Look at how it can be thrown:

https://github.com/openhab/openhab-core/blob/81f2bd9366d3f937da892ac704671207fce5e10c/bundles/org.openhab.core/src/test/java/org/openhab/core/library/types/DecimalTypeTest.java#L150

Reply:
I would suggest to change that in a separate PR because the same code is at least at two places. Here is another one:

https://github.com/openhab/openhab-core/blob/81f2bd9366d3f937da892ac704671207fce5e10c/bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/items/ItemUIRegistryImpl.java#L450